Improved DEVELOPERS.md (#5225)
supabase/supabase: DEVELOPERS.md

# Developing Supabase

* [Development Setup](##Development-Setup)
* [Installing Dependencies](###Installing-Dependencies)
* [Building Supabase](##Building-Supabase)
* [Start a Development Server](##Start-a-Development-Server)
* [What's Next?](##What's-Next)
* [Community Channels](##Community-Channels)

## Development Setup

First off, thanks for your interest in Supabase and for wanting to contribute!

Before you begin, read the [code of conduct](https://github.com/supabase/.github/blob/main/CODE_OF_CONDUCT.md) and check out the [existing issues](https://github.com/supabase/supabase/issues).

> 

Reviewer: Make this one paragraph.

Reviewer 2: Okay!

* [Installing Dependencies](#Installing-Dependencies)
* [Building Supabase](#Building-Supabase)
* [Start a Development Server](#Start-a-Development-Server)

> * [What's Next?](#What's-Next)

Reviewer: Can you change this to something else, the link doesn't work and needs to be changed.

Reviewer 2: Yup, I have changed and it works

To debug code, and to see changes in real time, it is often useful to have a local HTTP server. Click one of the three links below to choose which development server you want to start.

- [Supabase Website](###Supabase-Website-Development-Server)
- [Supabase Docs](###Supabase-Docs-Development-Server)

> - [Supabase Studio](###Supabase-Studio-Development-Server)

Reviewer: Any reason why change this?

Reviewer 2: Tbh, I don't know 😬 maybe while copy-pasting it might have changed. Sorry

Reviewer 2: I have re-added those

Reviewer: These changes break the links and need to be undone.

    http://localhost:8082/
    ```

> For more information on Supabase Studio, see the [Supabase Studio readme](./studio/README.md).

Reviewer: Let's not delete this, it leads to more in-depth docs on how to contribute to the studio.

Reviewer 2: Okay will add those back then

Make your changes to the file(s) you'd like to update.

### Open a pull request

When you're done making changes and you'd like to propose them for review by opening a pull request.

### Submit your PR & get it reviewed

- Once you submit your PR, others from the Unleash community will review it with you. The first thing you're going to want to do is a self review.
- After that, we may have questions, check back on your PR to keep up with the conversation.
- Did you have an issue, like a merge conflict? Check out GitHub's git tutorial on how to resolve merge conflicts and other issues.

### Your PR is merged!

Congratulations! The whole Supabase community thanks you 🙏🏼

> Once your PR is merged, you will be proudly listed as a contributor in the [contributor chart](https://github.com/supabase/supabase/graphs/contributors).

Reviewer: New change, let's not tell people how to open a pull request and issue. They can find that information elsewhere.

Reviewer 2: I suppose this information should be given especially to beginners (referred few organizations and their repos)

Reviewer 3: There is plenty of info available online on how to open a PR. Keep the doc short and succinct with only crucial info included. we shouldn't be assuming that readers are beginners.

Also, Unleash❓

## Build in Public
Once your PR is merged, you will be proudly listed as a contributor in the [contributor chart](https://github.com/supabase/supabase/graphs/contributors).

Videos of building supabase made in public.
## Community Channels

> - [Contributing to open source as  a beginner - Supabase](https://youtu.be/OAqhsylUuWg) - [@imskr](https://github.com/imskr)

Reviewer: Let's not delete this. This could be helpful for people that want to follow along.

Reviewer 2: Sure, we can have it

For more information on Supabase Studio, see the [Supabase Studio readme](./studio/README.md).

## What's Next?

Make your changes to the file(s) you'd like to update.

### Open a pull request

When you're done making changes and you'd like to propose them for review by opening a pull request.

### Submit your PR & get it reviewed

- Once you submit your PR, others from the Supabase community will review it with you. The first thing you're going to want to do is a self review.
- After that, we may have questions, check back on your PR to keep up with the conversation.
- Did you have an issue, like a merge conflict? Check out GitHub's git tutorial on how to resolve merge conflicts and other issues.

### Your PR is merged!

Congratulations! The whole Supabase community thanks you 🙏🏼

> Once your PR is merged, you will be proudly listed as a contributor in the [contributor chart](https://github.com/supabase/supabase/graphs/contributors).

Reviewer: Can you make this one paragraph

Reviewer: For example, you can make it something like this.

```suggestion
## Finally

After making your changes to the file(s) you'd like to update, it's time to open a pull request. Once you submit your pull request, others from the Supabase team/community will review it with you. 

Did you have an issue, like a merge conflict, or don't know how to open a pull request? Check out [GitHub's pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests) tutorial on how to resolve merge conflicts and other issues. Once your PR is merged, you will be proudly listed as a contributor in the [contributor chart](https://github.com/supabase/supabase/graphs/contributors).
```

Reviewer 2: Sure, let me update the PR
